### ISSUE TITLE
Templates / thumbnails

### ISSUE THREAD

**Issue by nbarlowATI** [alan-turing-institute/AssurancePlatform#44]
Could be nice for the homepage to offer a choice of template assurance cases, possibly via a big button with a thumbnail view of how the flowchart will look.
The templates themselves could be stored as JSON in this repo, loaded into CaseContainer, and then the new case saved in the DB as usual...
- [ ] design templates JSON
- [ ] create thumbnails
- [ ] figure out how to load into CaseContainer from JSON (have a prop with the JSON file path?)
- [ ] figure out how to save the resulting case as a new set of items in the DB (do we need a "save as" button?  Or just automatically save when the template is opened (need to save in cascade, from case, to goal, to children of goal, ...)

**Comment by mhauru** [comment]
Could we use the same JSON format for templates, import/export, and for how the backend returns the case information?

**Comment by mhauru** [comment]
I didn't have time to write any code for this today, but here's what I was thinking of doing:
1. Move `createDBObject` out of `ItemCreator.js` to some other source file, called something like `utilities` or whatnot (make all the state variables that it reads be arguments instead).
2. Write in that same file a function that goes through a JSON representation of an assurance case and posts it to the DB item by item.
3. In `CaseCreator.js`, add a selector for which template to use, with options being "empty" and all JSON files in some `templates` folder. Then when the case creation form is submitted, read the template JSON (if any) and call the function from 2 to add all elements from it to the newly created case.

This can then later be used to implemented import/export too: Export just dumps the JSON that the backend returns into a file, import reads a JSON file, creates a new case and calls the function from point 2 to add all the elements from the JSON into the case. After all, aren't importing a case and creating a new case based on a template essentially the same operation?

Actually, now that I wrote that, I started thinking, maybe the responsibilities of the function in 2. could be shifted onto the backend? The `POST` end-point of the backend could take a JSON reprensentation of a whole (sub)tree of an assurance case and write it all to the DB.
